containers/src/Kokkos_DynRankView.hpp

```diff
@@ -1100,20 +1100,9 @@ class DynRankView : public ViewTraits<DataType, Properties...> {
                    template computeRank<typename traits::array_layout, P...>(
                        arg_prop, arg_layout)) {
     // Append layout and spaces if not input
-    using alloc_prop_input = Kokkos::Impl::ViewCtorProp<P...>;
-
-    // use 'std::integral_constant<unsigned,I>' for non-types
-    // to avoid duplicate class error.
-    using alloc_prop = Kokkos::Impl::ViewCtorProp<
-        P...,
-        std::conditional_t<alloc_prop_input::has_label,
-                           std::integral_constant<unsigned, 0>, std::string>,
-        std::conditional_t<alloc_prop_input::has_memory_space,
-                           std::integral_constant<unsigned, 1>,
-                           typename traits::device_type::memory_space>,
-        std::conditional_t<alloc_prop_input::has_execution_space,
-                           std::integral_constant<unsigned, 2>,
-                           typename traits::device_type::execution_space>>;
+    using alloc_prop = decltype(Impl::add_properties(
+        arg_prop, std::string{}, typename traits::device_type::memory_space{},
+        typename traits::device_type::execution_space{}));
```

Comment:
How about
```suggestion
    // Copy the input allocation properties with possibly defaulted properties
    auto prop_copy = Impl::add_properties(
        arg_prop, std::string{}, typename traits::device_type::memory_space{},
        typename traits::device_type::execution_space{});
    using alloc_prop = decltype(prop_copy);
```

It seems to read more cleanly to actually call the function, and then query the type of the resulting object, than to come up with the type based on a hypothetical call, and construct an instance of that

Reply:
Most of the other call sites are already doing this